src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,49 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines;
+        if (type.equals(Bulldozer.class)) {
+            machines = new BulldozerProducer().get();
+            return new ArrayList<>(machines);
```

Comment:
```suggestion
        if (type.equals(Bulldozer.class)) {
            return new ArrayList<>(bulldozerProducer().get());
```
lets create class-level producers, this way you wont create a new instance each time this method is invoked
ps remember to you abstract type as a field type